### DISCUSSION TITLE
2-way sync with local folder

### DISCUSSION THREAD

**Discussion by andreaferretti** [ente-io/ente#1318] [Enhancements]
Right now, we have two separate 1-way syncs:
* we can watch a folder and upload images in this folder to Ente
* or we can download Ente files to a folder

Is there/will ever be there the possibility to just 2-way sync with a local folder, much like Dropbox/NextCloud and similar? This is kind of a blocker for me, since I don't want to have my data just in cloud

**Comment by mnvr** [comment]
> since I don't want to have my data just in cloud

The export supports a continuous sync - https://help.ente.io/photos/migration/export/#sync-continuously - so you can just keep it running in the background and it'll automatically export when things change within your Ente Photos app, so you will have a local copy of your data that is not tied to the cloud.

There is also a CLI tool you can use to export, say on a cron job - https://help.ente.io/photos/faq/export#can-i-backup-my-data-in-a-local-drive-outside-ente

**Reply by henrivdm** [reply]
This is a messy work-around - if you use any other software on a file you need to reimport, which is messy:  good example: if you just add a tag or description to a photo in windows, you would need to remember to move (?) it from the Export folder to a watchfolder.

Ente is shooting themselves in the foot to dig their heels in on this:  a proper 2-way synch would make Ente a Google Photo and Onedrive killer, because it will represent the features missing from them respectively.

**Comment by andreaferretti** [comment]
Thank you for your answer! It is still not ideal that the 2 way syncs happen in separate folders, so if you watch a folder and activate sync continously, you will get duplicated files, but it is good to know that there is a way to keep all my files locally!

**Comment by fronglemog** [comment]
I haven't found any comment on this yet; if I delete a photo from a watched folder, is it deleted from ente?

I think continuous export works like this in the other direction, but I haven't figured it out for watched folders.

**Reply by mnvr** [reply]
The intended behaviour is that if a file is deleted locally, then the corresponding Ente file will also be automatically moved to uncategorized.

Unfortunately, the current version (1.7.0) has a bug where this is not happening automatically, which is likely why you got confused. Apologies. I've fixed that bug (https://github.com/ente-io/ente/pull/2092), and the fix will be out in the next release.

> The fix is already in the [nightly build](https://github.com/ente-io/photos-desktop/releases) if you really need it asap, but as usual all the caveats about the mainline nightly builds apply, so only do this if you know what you're doing.

Thank you for bringing this up.

**Reply by reitzig** [reply]
How is "move to uncategorized" an implementation of "delete from ente"? :thinking:

**Comment by StefanOltmann** [comment]
Yes, having real 2-way-sync would be great.

I want to replace Resilio Sync for my photos with Ente.

**Comment by Leonardoes22** [comment]
Although Ente seems really promising, there are still many features missing. A great benefit of having 2-way sync is the posibility to organize our photos with an external program or script when these features are not implemented yet in Ente.

This would allow the bulk of organization to be done in Ente, but still allow some edge cases to be done outside of it.

**Comment by henrivdm** [comment]
I am afraid I cannot make the jump from Onedrive and Google Photos to Ente unless proper 2-way synch and tag-embedding is in place.  I would LOVE to make the move!

**Comment by henrivdm** [comment]
see also #7490
